src/main/cpp/src/CastStringJni.cpp

```diff
@@ -111,4 +118,72 @@ JNIEXPORT jlong JNICALL Java_com_nvidia_spark_rapids_jni_CastStrings_fromDecimal
   }
   CATCH_CAST_EXCEPTION(env, 0);
 }
+
+JNIEXPORT jlong JNICALL Java_com_nvidia_spark_rapids_jni_CastStrings_toIntegersWithBase(
+  JNIEnv* env, jclass, jlong input_column, jint base)
+{
+  JNI_NULL_CHECK(env, input_column, "input column is null", 0);
+  using namespace cudf;
+  try {
+    jni::auto_set_device(env);
+    auto const res_data_type = data_type(type_id::UINT64);
+
+    auto const input_view{*reinterpret_cast<column_view const*>(input_column)};
+    const strings_column_view input_strings{input_view};
+    auto integer_view = [&] {
+      switch (base) {
+        case 10: {
+          // TODO implement it in the kernel
+          auto const regex = strings::regex_program::create(R"(^\s*([0-9]+).*)");
```

Comment:
interestingly I started out with strip but dropped because of the hex_to_integers behavior but it might indeed work with the decimal str to_integers.

```Python
>>> import cudf
>>> gsr = cudf.Series('5g5')
>>> cudf._lib.string_casting.htoi(gsr._column)
<cudf.core.column.numerical.NumericalColumn object at 0x7f6dc54d3ec0>
[
  85
]
dtype: int64
>>> cudf._lib.string_casting.stoi(gsr._column)
<cudf.core.column.numerical.NumericalColumn object at 0x7f6dc54d3f40>
[
  5
]
dtype: int32
>>> 
```